text/0085-multi-token-standard.md

```diff
@@ -1,87 +1,210 @@
-# Title
+# Casper Multi-Token Standard 
+CEP PR: casperlabs/ceps#0085
 
 ## Summary
+A standard that enables multiple fungible and non-fungible tokens to exist within a single smart contract instance on a Casper network.
 
-[summary]: #summary
+## Motivation
+We aim to provide a reference for the Multi Token Standard that in design and quality that should be adhered to for optimal execution of contracts upon the Casper blockchain system.
 
-CEP PR: [casperlabs/ceps#0000](https://github.com/casperlabs/ceps/pull/0000)
+A Multi-Token standard enables functionality that cannot be served by Fungible and Non-Fungible token standards separately. Creating a smart contract that combines these two features brings Casper in line with other top-tier blockchain platforms, and bolsters the use cases available on Casper networks. This standard ostensibly allows for limited-supply fungible tokens with added information, i.e., digital admission tickets, open-edition art and so on.
 
-One paragraph explanation of the feature.
+## Guide-level Explanation
 
-## Motivation
+### Interface
+
+Any compliant contract should contain the following endpoints:
+
+#### balance_of
+
+Returns the amount of a specified token that a given address holds.
+
+```rust
+fn balance_of(account: Key, id: U256) -> U256
+```
+
+#### balance_of_batch
+
+Returns the balance of a batch of accounts for a specified token.
+
+```rust
+fn balance_of_batch(accounts: Vec<Key>, ids: Vec<U256) -> Vec<U256>
+```
+
+#### mint
+
+Mints an amount of the specified token and places it in the balance of the `recipient`.
+
+```rust
+fn mint(recipient: Key, id: U256, amount: U256)
+```
+
+#### batch_mint
+
+Batch mints the provided amount of multiple tokens and places them in the balance of the `recipient`.
+
+```rust
+fn batch_mint(recipient: Key, ids: Vec<U256>, amounts: Vec<U256>)
+```
```

Comment:
Don't forget those entrypoints are called by cli interface the casper-client or an sdk. You don't want to complexity the arguments more than that for users, tupling things can lead to complex args building, despite we have Tuple2/3 in Casper types, it is still a complex arg type for some users.  The erc is built that way https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#ERC1155-_mintBatch-address-uint256---uint256---bytes- so basically it is the kind of behavior we change when we have a gain in either cost or usage. Here it's pretty standard to specify an id an an amount coupled that way, not very rusty way indeed, that is also how users of this kind of nft/token contract are used to interact with and we don't want to change their habits too much otherwise usually feedback is coming back that this is not proper standard compared to others.

Reply:
Since the people writing these contracts and reading these docs are expected to know basic Rust, I don't think using tuples (which exist in many languages, like Python) are too much. An example can even be provided using zip (for pairs) or itertools::izip! (for n-ary tuples). That said, in the interests of moving on, I'll drop my request for change and accept this.